sklearn/neural_network/tests/test_elm.py

```diff
@@ -0,0 +1,393 @@
+    """Test partial_fit for classification.
+
+    It should output the same results as 'fit' for binary and
+    multi-class classification.
+    """
+    for X, y in classification_datasets.values():
+        batch_size = 100
+        n_samples = X.shape[0]
+
+        elm_fit = ELMClassifier(random_state=random_state,
+                                batch_size=batch_size)
+        elm_partial_fit = ELMClassifier(random_state=random_state)
+
+        elm_fit.fit(X, y)
+        for batch_slice in gen_batches(n_samples, batch_size):
+            elm_partial_fit.partial_fit(X[batch_slice], y[batch_slice],
+                                        classes=np.unique(y))
+
+        pred1 = elm_fit.predict(X)
+        pred2 = elm_partial_fit.predict(X)
+
+        assert_array_equal(pred1, pred2)
+        assert_greater(elm_fit.score(X, y), 0.95)
+        assert_greater(elm_partial_fit.score(X, y), 0.95)
+
+
+def test_partial_fit_regression():
+    """Test partial_fit for regression.
+
+    It should output the same results as 'fit' for regression on
+    different activations functions.
+    """
+    X = Xboston
+    y = yboston
+    batch_size = 100
+    n_samples = X.shape[0]
+
+    for activation in ACTIVATION_TYPES:
+        elm_fit = ELMRegressor(random_state=random_state, C=100,
+                               activation=activation, batch_size=batch_size)
+
+        elm_partial_fit = ELMRegressor(activation=activation,
+                                       C=100,
+                                       random_state=random_state,
+                                       batch_size=batch_size)
+
+        elm_fit.fit(X, y)
+        for batch_slice in gen_batches(n_samples, batch_size):
+            elm_partial_fit.partial_fit(X[batch_slice], y[batch_slice])
+
+        pred1 = elm_fit.predict(X)
+        pred2 = elm_partial_fit.predict(X)
+
+        assert_almost_equal(pred1, pred2, decimal=2)
+        assert_greater(elm_fit.score(X, y), 0.85)
+        assert_greater(elm_partial_fit.score(X, y), 0.85)
+
+
+def test_predict_proba_binary():
+    """Test whether predict_proba works as expected for binary class."""
+    X = Xdigits_binary[:50]
+    y = ydigits_binary[:50]
+
+    clf = ELMClassifier(n_hidden=10)
+    clf.fit(X, y)
+    y_proba = clf.predict_proba(X)
+
+    (n_samples, n_classes) = y.shape[0], 2
+
+    assert_equal(y_proba.shape, (n_samples, n_classes))
+    assert_greater(roc_auc_score(y, y_proba[:, 1]), 0.95)
+
+
+def test_predict_proba_multi():
+    """Test whether predict_proba works as expected for multi class."""
+    X = Xdigits_multi[:10]
+    y = ydigits_multi[:10]
+
+    clf = ELMClassifier(n_hidden=5)
+    clf.fit(X, y)
+    y_proba = clf.predict_proba(X)
+
+    (n_samples, n_classes) = y.shape[0], np.unique(y).size
+
+    assert_equal(y_proba.shape, (n_samples, n_classes))
+
+
+def test_recursive_and_standard():
+    """Test that recursive lsqr return the same result as standard lsqr."""
```

Comment:
Does this test something else than the partial_fit test above? Well I guess more tests can't hurt.